Allow choice in packaged units for selecting high or low speed air flow for no load supply air flow rate

rraustad opened the pull request:
Pull request overview
---------------------
 - Fixes #10312
 - Correct parent object No Load Supply Air Flow Rate for Coil:Cooling:DX. Include other coil objects that have multiple air flow rates. Allow user to select either high or low speed used for no load air flow rate. Applies to AirloopHVAC:UnitarySystem, ZoneHVAC:PackagedTerminalAirConditioner, ZoneHVAC:PackagedTerminalHeatPump and ZoneHVAC:WaterToAirHeatPump.
 - Fixes #10380 where packaged units have a blank input for No Load Supply Air Flow Rate and the fan operating mode schedule specifies constant fan and a Fan:SystemModel object is used where the air flow rate would be proportional to coil PLR instead of operating at the full operating air flow rate.
 
ToDo:
 - [x] add other VS coils to logic
 - [x] discuss method to allow user to select low or high speed when autosizing no load air flow rate
 - [x] example file updates
 - [x] transition code
 - [x] expand objects code
 - [x] update docs
 - [x] unit test

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

rraustad committed:
Correct new DX coil no load supply air flow rate

rraustad commented:
I am considering adding an input field just after the no load input field for parent objects. Thoughts?

    ZoneHVAC:PackagedTerminalAirConditioner,
      N3, \field No Load Supply Air Flow Rate
          \type real
          \units m3/s
          \minimum 0
          \autosizable
          \note Must be less than or equal to fan size.
          \note Only used when supply air fan operating mode schedule values specify continuous fan
          \note (schedule values greater than 0 specify continuous fan operation).
          \note This air flow rate is used when no heating or cooling is required and the cooling or
          \note heating coil is off. If this field is left blank or zero, the supply air flow rate
          \note from the previous on cycle (either cooling or heating) is used.
      A7, \field Use Low Speed Coil Air Flow Rate As No Load Supply Air Flow Rate
          \type choice
          \key Yes
          \key No
          \default No
          \note When Yes is selected the minimum air flow rate is used.
          \note When No is selected the maximum air flow rate is used.

mjwitte commented:
> I am considering adding an input field just after the no load input field for parent objects. Thoughts?
> 
> ```
> ZoneHVAC:PackagedTerminalAirConditioner,
>   N3, \field No Load Supply Air Flow Rate
>       \type real
>       \units m3/s
>       \minimum 0
>       \autosizable
>       \note Must be less than or equal to fan size.
>       \note Only used when supply air fan operating mode schedule values specify continuous fan
>       \note (schedule values greater than 0 specify continuous fan operation).
>       \note This air flow rate is used when no heating or cooling is required and the cooling or
>       \note heating coil is off. If this field is left blank or zero, the supply air flow rate
>       \note from the previous on cycle (either cooling or heating) is used.
>   A7, \field Use Low Speed Coil Air Flow Rate As No Load Supply Air Flow Rate
>       \type choice
>       \key Yes
>       \key No
>       \default No
>       \note When Yes is selected the minimum air flow rate is used.
>       \note When No is selected the maximum air flow rate is used.
> ```

Would the new field be ignored when N3 is blank?
If the system has different low speed flow rates for heating and cooling, then what gets used?

Another approach is to make the new field be something like
```
A7, \field No Load Supply Air Flow Rate Control
       \type choice
       \key SpecifiedFlowRate (N3)
       \key MinimumFlowRate
       \key MaximumFlowRate
       \key PreviousFlowRate (same as when N3 is blank currently)
       \key ????
       \default ???
```

rraustad commented:
>Would the new field be ignored when N3 is blank?

I'll have to look into what the blank actually does but I think it would be independent of this new input.

> If the system has different low speed flow rates for heating and cooling, then what gets used?

the minimum of the 2

Good suggestion for a choice list. This will probably only be used if N3 is autosized but I will check on that. If anything other than Yes/No is needed I'll go with a list.

rraustad committed:
Add other coil types

rraustad committed:
Add new input field

rraustad committed:
Update unit tests

rraustad committed:
Update example files

Nigusse commented:
@rraustad is the no-load airflow rate tied to the minimum ventilation air requirement?

rraustad commented:
> @rraustad is the no-load airflow rate tied to the minimum ventilation air requirement?

The minimum supply air flow should be greater than or equal to the OA flow rate but that should be handled elsewhere. That would be a separate pull request if it's not happening now. I am still working on all other coils so the user has an option of picking how to set no load air flow rate.

rraustad commented on src/EnergyPlus/UnitarySystem.cc:
```diff
@@ -2114,25 +2114,75 @@ namespace UnitarySystems {
             } else {
-                if (this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed ||
-                    this->m_HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed) {
-                    if (this->m_CoolCoilExists && this->m_CoolingCoilType_Num == DataHVACGlobals::Coil_CoolingAirToAirVariableSpeed) {
```
This check only happened for VS DX coil types. This change opens up the ability to set no load air flow rate for all other multi/variable-speed coils. The user can now choose how to set no load air flow rate by choosing either the low or high speed air flow rate. I hope I got all the coil types here that can have multiple air flow rates.

rraustad committed:
Update example files

rraustad committed:
Fix some mistakes adding new field

rraustad committed:
Update 2 more example files

rraustad committed:
Update min fields and add note, update ExpandObjects, update transition rules and code.

rraustad committed:
Update docs and IDD notes

rraustad committed:
Update more example files

rraustad committed:
Correct 1 example file to show no diffs

rraustad commented on src/Transition/CreateNewIDFUsingRulesV24_1_0.f90:
```diff
@@ -382,7 +382,18 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
+                  OutArgs(38)=InArgs(38)
+                  ! OutArgs(39) is added. If existing coil is VS DX cooling or heating coil then set YES, otherwise set NO
+                  IF ( SameString(InArgs(12),'Coil:Heating:DX:VariableSpeed') .or. SameString(InArgs(15),'Coil:Cooling:DX:VariableSpeed') ) THEN
+                      OutArgs(39) = 'YES'
```
@mjwitte does this argument need to be lower case "Yes" to match the idd?

    A7 , \field No Load Supply Air Flow Rate Control Set To Low Speed
        \type choice
        \key Yes
        \key No

mjwitte replied:
> @mjwitte does this argument need to be lower case "Yes" to match the idd?

Not required, but looks nicer.

rraustad commented on src/Transition/CreateNewIDFUsingRulesV24_1_0.f90:
```diff
@@ -434,6 +445,37 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
+                  CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                  nodiff=.false.
+                  OutArgs(9)=InArgs(9)
+                  ! OutArgs(39) is added. If existing coil is VS DX cooling or heating coil then set YES, otherwise set NO
```
The comment is wrong here and below, should be OutArgs(10) is added.

rraustad committed:
Fix and test transition program and add checks in unit tests

rraustad committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into 10312-No-Load-Supply-Air-Flow-Rate-using-VS-coils

rraustad commented:
Fully tested transition program with all 4 objects (PTAC, PTHP, WSHP, UnitarySystem), with and without VS DX coil, and all now transition correctly (i.e., Yes if VS, No if not). This is ready for review of code, variable names, input field names, docs, etc.

rraustad commented:
Defect file result:

    Component Sizing Information, ZoneHVAC:PackagedTerminalAirConditioner, ZONE2PTAC, Design Size Cooling Supply Air Flow Rate [m3/s], 0.21487
    Component Sizing Information, ZoneHVAC:PackagedTerminalAirConditioner, ZONE2PTAC, Design Size Heating Supply Air Flow Rate [m3/s], 0.21487
    Component Sizing Information, ZoneHVAC:PackagedTerminalAirConditioner, ZONE2PTAC, Design Size No Load Supply Air Flow Rate [m3/s], 7.73521E-002
    Component Sizing Information, Coil:Cooling:DX:CurveFit:Speed, PTAC 2 DX COOL COOLING COIL SPEED 1 PERFORMANCE, User-Specified Rated Air Flow Rate [m3/s], 7.73521E-002

[PackagedTerminalAirConditionerVSAS_NewCoilModel-Updated.idf.txt](https://github.com/NREL/EnergyPlus/files/13680177/PackagedTerminalAirConditionerVSAS_NewCoilModel-Updated.idf.txt)

nrel-bot-2b commented:
@mjwitte @rraustad @myoldmopar it has been 28 days since this pull request was last updated.

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into 10312-No-Load-Supply-Air-Flow-Rate-using-VS-coils

mjwitte commented:
@rraustad Tested transition on the defect file attached to the issue (PTAC with Coil:Cooling:DX) and compared the new field "No Load Supply Air Flow Rate Control Set To Low Speed" with Yes vs No. 

eio sizing results change as expected. 

Simulation flow results for PTAC3 look reasonable (this has cycling fan at night, continuous during the day)
![image](https://github.com/NREL/EnergyPlus/assets/4991075/032f81f6-4d89-4a80-b839-1987adb439fd)

PTAC2 seems to have no change, because this uses cycling fan. Because this is detailed output, there's a slight shift in number of rows, so ignore the alignment in the chart, but why would there be any change at all? Oh, maybe because conditions in other zones are changing?
![image](https://github.com/NREL/EnergyPlus/assets/4991075/16fade9f-e771-4f82-903b-d7d680226b20)

But PTAC1 (which has blank for No Load Supply Air Flow Rate and has continuous fan 24/7) has higher flow rates with Yes than with No at certain times?
![image](https://github.com/NREL/EnergyPlus/assets/4991075/27511852-4dfa-4c87-a6a6-04f2835938c6)

rraustad commented:
The only issue I see is Zone1PTAC. A change in detailed timestep would result from any zone changing temperature more than the tolerance. So even if 1 zone has a cycling coil and wasn't expected to change operation, other zones could force a change in detailed time step length. Eliminating the shift in the data, those traces for Zone2PTAC are right on top of each other. The Zone1PTAC has Coil:Cooling:DX, hard-sized capacity and flow rates, and a blank no load and no load control inputs (you are testing Yes and No here with blank defaulting to Yes). I have to think about what all that means, if anything at all. Speed 1 air flow fraction = 0.36, I would expect that to yield a 0.4 x 0.36 = 0.144 m3/s no load flow rate. I assume the mass flow rate is showing this value (0.144 m3/s ~ 0.17 kg/s). That's the good part of the figure, and appears to be during the day so using constant fan. There must be no load there for the PTAC to be running at the no load air flow rate. For night cycling coil operation, I would expect a load since the fan is on, and a different load for each test of Yes and No. Do you have the zone temp to compare? Tstat heating = 22.5 all day. So hopefully the zone temp explains this. Otherwise I'll have to check for a control issue. I agree this does seem odd. Zone1PTAC is hardsized so maybe it's a capacity issue. Oh wait, Zone1PTAC has an OA mixer, I wonder if Zone1PTAC is bringing in a different amount of outdoor air? Or maybe both things are happening, where the zone temps are different causing the PTAC to go full flow to meet the load, brings in max OA, can't get there and runs all night like that. Even trying to think this through without running the file leaves me wondering what is happening. Show the temps for the zone and then I'll take it from there.

    ZoneHVAC:PackagedTerminalAirConditioner,
      Zone1PTAC,               !- Name
      FanAndCoilAvailSched,    !- Availability Schedule Name
      Zone1PTACAirInletNode,   !- Air Inlet Node Name
      Zone1PTACAirOutletNode,  !- Air Outlet Node Name
      OutdoorAir:Mixer,        !- Outdoor Air Mixer Object Type
      Zone1PTACOAMixer,        !- Outdoor Air Mixer Name
      0.4,                     !- Cooling Supply Air Flow Rate {m3/s}
      0.4,                     !- Heating Supply Air Flow Rate {m3/s}
      ,                        !- No Load Supply Air Flow Rate {m3/s}
      ,                        !- No Load Supply Air Flow Rate Control Set To Low Speed

    Coil:Cooling:DX:CurveFit:OperatingMode,
      PTAC 1 DX Cool Cooling Coil Operating Mode,  !- Name
      7200,                    !- Rated Gross Total Cooling Capacity {W}
      0.4,                     !- Rated Evaporator Air Flow Rate {m3/s}


    Coil:Cooling:DX:CurveFit:Speed,
      PTAC 1 DX Cool Cooling Coil Speed 1 Performance,  !- Name
      0.225525303,             !- Gross Total Cooling Capacity Fraction
      0.36,                    !- Evaporator Air Flow Rate Fraction

rraustad commented:
It doesn't look like the zone temp explains this 

![image](https://github.com/NREL/EnergyPlus/assets/5778156/7fe5b445-92c6-4693-9dde-32bc689149fc)

rraustad commented:
@mjwitte Zone1PTAC has a constant fan schedule. That does explain things. Agreed?

rraustad commented:
Zone1PTAC runs at the max flow in heating mode because the heating coil is on and then is running at the low speed cooling flow in cooling mode again becuase the coil is on, in winter.

rraustad commented:
Then why didn't the Zone1PTAC heating coil run at full flow in heating mode with the no load flow control set to No?

rraustad committed:
Fix issue with blank no load supply air flow rate for PTUnit

rraustad commented:
I see what's happening. Zone1PTAC is the only one that has a blank input for No Load Supply Air Flow Rate (and a very good thing the test file had this input). That blank input sets a flag when the No Load Supply Air Flow Rate field is blank (or input as 0), as `this->m_AirFlowControl = UseCompFlow::On`. And only when there is a Supply Air Fan Operating Mode Schedule Name present since the PTUnits default to cycling fan if this field is blank. Also only for Fan:System model since it's the only fan that uses the ratios described later.

        if (this->m_FanOpModeSchedPtr > 0) {
            if (!ScheduleManager::CheckScheduleValueMinMax(state, this->m_FanOpModeSchedPtr, ">=", 0.0, "<=", 0.0)) {
                //           set fan operating mode to continuous so sizing can set VS coil data
                this->m_FanOpMode = DataHVACGlobals::ContFanCycCoil;
                //           set air flow control mode:
                //             UseCompressorOnFlow = operate at last cooling or heating air flow requested when compressor is off
                //             UseCompressorOffFlow = operate at value specified by user
                //           AirFlowControl only valid if fan opmode = ContFanCycComp
                if (this->m_MaxNoCoolHeatAirVolFlow == 0.0) {
                    this->m_AirFlowControl = UseCompFlow::On;
                } else {
                    this->m_AirFlowControl = UseCompFlow::Off;
                }
            }
        }

And then later during initialization, for PTUnits only (a hack to minimize diffs during the transition to UnitarySystem), the m_MaxNoCoolHeatAirVolFlow was reset for this control. But the `m_NoLoadAirFlowRateRatio` is now calculated for this coil type in the new code I added, and was never reset here for this control. So what you saw for Zone1PTAC was the fan operating like it was a cycling fan because the system fan model uses the on and off flow ratio (passed to the fan model) in the calculation of flow. Here noFlowRatio was 0 (because the no flow field was blank, which is initialized to 0 in the constructor, and the new code calculation of noFlowRatio resulted in a 0) when it should have been 1. This problem (blank or 0 no flow input with constant fan has proportional air flow to PLR) was likely happening for the VS coil before this code changed (because VS coil was the only coil type doing this math and I expanded to other multi speed coil types) when a system fan was used. PR description needs this update.

        if (this->m_sysType == SysType::PackagedAC || this->m_sysType == SysType::PackagedHP || this->m_sysType == SysType::PackagedWSHP) {
            if (this->m_AirFlowControl == UseCompFlow::On) {
                this->m_MaxNoCoolHeatAirVolFlow = min(this->m_MaxCoolAirVolFlow, this->m_MaxHeatAirVolFlow);
            }

I added to this block (which hopefully won't change any other example file results):

            if (this->m_AirFlowControl == UseCompFlow::On) {
                this->m_MaxNoCoolHeatAirVolFlow = min(this->m_MaxCoolAirVolFlow, this->m_MaxHeatAirVolFlow);
                this->m_NoLoadAirFlowRateRatio = 1.0;   <--- new code to fix this problem
            }

And now all PTUnits look the same.

Notice here that the Zone1PTAC heating coil heating rate looks like the same shape as the air mass flow rate you saw before this change. Now that the code was corrected the air flow rate shows a constant fan mode flow rate.

![image](https://github.com/NREL/EnergyPlus/assets/5778156/a4f49301-b870-4832-b74e-cc4a8770012b)

For the cooling design day, there is no change, while Zone2PTAC and Zone3PTAC no flow supply air flow rate DID change in the eio. I suspect that the coil is operating (i.e., there is a load) and zone temps seem to reflect that.

![image](https://github.com/NREL/EnergyPlus/assets/5778156/8c138609-1e5a-4652-819e-3c1e54f8defc)

rraustad commented on src/EnergyPlus/UnitarySystem.cc:
```diff
@@ -2114,25 +2114,75 @@ namespace UnitarySystems {
                                 state.dataVariableSpeedCoils->VarSpeedCoil(this->m_CoolingCoilIndex).MSRatedAirVolFlowRate(1) / MaxSpeedFlowRate;
+                        } else {
+                            // I think all these little IFs need an else in case these inputs are autosized proportional to number of speeds/stages
+                            NoLoadCoolingAirFlowRateRatio = state.dataVariableSpeedCoils->VarSpeedCoil(this->m_CoolingCoilIndex).NumOfSpeeds;
```
This needs to be 1 / NumOfSpeeds. Maybe these `} else {` statements aren't even needed since if flow = 0 then nothing would happen anyway.

rraustad replied:
I think I had to add these `} else {` to cover autosized inputs where MaxSpeedFlowRate = -99999 here before the coil is sized. So yeah, this does need to change.

rraustad replied:
These are the speed air flows, and they are allowed to be 0. So this does look OK.

    Coil:Cooling:DX:VariableSpeed,
    N19, \field Speed 1 Reference Unit Rated Air Flow Rate
        \units m3/s
        \type real
        \minimum 0
        \required-field

    N91, \field Speed 10 Reference Unit Rated Air Flow Rate
        \units m3/s
        \type real
        \minimum 0

rraustad committed:
Correct ratio calculation and change new var name.

rraustad commented on src/EnergyPlus/UnitarySystem.cc:
```diff
@@ -2114,25 +2114,75 @@ namespace UnitarySystems {
                         if (MaxSpeedFlowRate > 0.0) {
                             NoLoadHeatingAirFlowRateRatio =
                                 state.dataVariableSpeedCoils->VarSpeedCoil(this->m_HeatingCoilIndex).MSRatedAirVolFlowRate(1) / MaxSpeedFlowRate;
                         }
```
What are the odds that a user enters a 0 for an air flow field? I missed this one when adding the `} else {` to all these coil types.

rraustad commented on src/EnergyPlus/UnitarySystem.cc:
```diff
@@ -2114,25 +2114,75 @@ namespace UnitarySystems {
+                                                      .MSRatedAirVolFlowRate(state.dataDXCoils->DXCoil(this->m_CoolingCoilIndex).NumOfSpeeds);
+                        if (MaxSpeedFlowRate > 0.0) {
+                            NoLoadCoolingAirFlowRateRatio =
+                                state.dataDXCoils->DXCoil(this->m_CoolingCoilIndex).MSRatedAirVolFlowRate(1) / MaxSpeedFlowRate;
```
What if the user does something crazy and does not autosize the high speed air flow rate but does autosize the low speed air flow rate?

rraustad committed:
More protection on ratio calculation and docs

rraustad commented:
@mjwitte I am done making changes.

Myoldmopar commented:
Wow, this was a fun read!  I think @rraustad settled your concerns @mjwitte , but you haven't commented to confirm.  Could you take a quick peek at the more recent responses and see if you are satisfied?

Myoldmopar commented on doc/input-output-reference/src/overview/group-unitary-equipment.tex:
```diff
@@ -297,6 +297,10 @@ \subsubsection{Inputs}\label{inputs-049}
+\paragraph{Field: No Load Supply Air Flow Rate Control Set To Low Speed}\label{field-no-load-supply-air-flow-rate-control-set-to-low-speed-004}
+
+This alpha field defines whether the supply air flow rate leaving the air conditioner in cubic meters per second when neither cooling nor heating is required is set to the coil high speed or low speed air flow rate. The valid choices are Yes or No. The default value is Yes. If the coil type does not have multiple air flow rates or if Design Specification Multispeed Object Type = UnitarySystemPerf ...
+
```
New field description makes sense.

Myoldmopar commented on performance_tests/15zonePTAC.idf:
```diff
@@ -2031,6 +2031,7 @@
     AUTOSIZE,                !- Cooling Supply Air Flow Rate {m3/s}
     AUTOSIZE,                !- Heating Supply Air Flow Rate {m3/s}
     AUTOSIZE,                !- No Load Supply Air Flow Rate {m3/s}
+    ,                        !- No Load Supply Air Flow Rate Control Set To Low Speed
```
Whew, lots of these!

Myoldmopar commented on src/Transition/CreateNewIDFUsingRulesV24_1_0.f90:
```diff
@@ -434,6 +447,38 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
+                  ! Coil:*:WaterToAirHeatPump:VariableSpeedEquationFit was not previously used to set no load air flow rate
+                  OutArgs(10)='No'
+                  OutArgs(11:CurArgs+1)=InArgs(10:CurArgs)
+                  CurArgs = CurArgs + 1
```
These changes seem reasonable.

Myoldmopar commented on tst/EnergyPlus/unit/PackagedTerminalHeatPump.unit.cc:
```diff
@@ -4897,6 +4908,7 @@ TEST_F(EnergyPlusFixture, PTAC_AvailabilityManagerTest)
     EXPECT_EQ("ZoneHVAC:PackagedTerminalAirConditioner", thisSys.UnitType);
     EXPECT_EQ("COIL:HEATING:FUEL", thisSys.m_HeatingCoilTypeName);
     EXPECT_EQ(state->dataHeatingCoils->HeatingCoil(1).HCoilType_Num, Coil_HeatingGasOrOtherFuel);
+    EXPECT_FALSE(thisSys.m_useNoLoadLowSpeedAirFlow);
```
Checks input, this is fine.

Myoldmopar commented on tst/EnergyPlus/unit/UnitarySystem.unit.cc:
```diff
@@ -1066,6 +1069,7 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_TwoSpeedDXCoolCoil_Only)
     EXPECT_NEAR(state->dataLoopNodes->Node(2).Temp, state->dataLoopNodes->Node(2).TempSetPoint, 0.001);
     // cooling coil air inlet node temp is greater than cooling coil air outlet node temp
     EXPECT_GT(state->dataLoopNodes->Node(3).Temp, state->dataLoopNodes->Node(2).Temp);
+    EXPECT_TRUE(thisSys->m_useNoLoadLowSpeedAirFlow);
```
I didn't see any unit tests that actually test the operation, just ensuring this flag is being set properly.  I won't hold this up for that, but in a future effort on here, maybe you could extract out the operation into a worker function that can be easily tested on its own.

Myoldmopar reviewed (approved):
Code changes look good.

Myoldmopar commented:
CI looks happy, just table diffs.  @mjwitte if you are satisfied here, merge away.  Or let me know and I can.  Thanks @rraustad and @mjwitte !

mjwitte reviewed (approved):
@rraustad Re-tested this and PTAC1 looks great. Thanks for the additional fix. Merging.
